uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1018.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_AbsenceType as AbsenceType;
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+use CRM_HRLeaveAndAbsences_Service_ContactWorkPattern as ContactWorkPattern;
+
+trait CRM_HRLeaveAndAbsences_Upgrader_Step_1018 {
+
+  /**
+   * As per PCHR-3164 end time can be chosen for single day leave requests
+   * in hours calculation unit. However, existing such leave requests are
+   * storing end time currently equal to start time. This upgrader fetches such
+   * leave requests, retrieves information about work pattern day basing on the
+   * date of the leave request and, if it is still a working day, sets the end
+   * time from the working day as an end time of the leave request.
+   *
+   * @return boolean
+   */
+
+  public function upgrade_1018 () {
+    $calculationUnits = array_flip(AbsenceType::buildOptions('calculation_unit', 'validate'));
+
+    $leaveRequestTable = LeaveRequest::getTableName();
+    $absenceTypeTable = AbsenceType::getTableName();
+
+    $query = "SELECT leaveRequest.id
+      FROM {$leaveRequestTable} leaveRequest
+      INNER JOIN {$absenceTypeTable} absenceType ON leaveRequest.type_id = absenceType.id
+      WHERE
+        absenceType.calculation_unit = {$calculationUnits['hours']}
```

Comment:
Can you move this line up with the WHERE statement and let the other lines align.

Reply:
Done.